electron-app/src/server/websites/wordpress/wordpress.service.ts

```diff
@@ -0,0 +1,231 @@
+    } else {
+      status.loggedIn = true;
+      status.username = accountInfo.name
+    }
+
+    return status;
+  }
+
+  getScalingOptions(file: FileRecord): ScalingOptions {
+    return { maxSize: FileSize.MBtoBytes(50) };
+  }
+
+  async postNotificationSubmission(
+    cancellationToken: CancellationToken,
+    data: PostData<Submission, WordPressNotificationOptions>,
+    accountData: WordPressAccountData,
+  ): Promise<PostResponse> {
+    this.checkCancelled(cancellationToken);
+    const res = await Http.post<any>(accountData.instance + '/wp-json/wp/v2/posts', '', {
+      data: this.buildDescriptionPayload(data, accountData, null),
+      type: 'multipart',
+      skipCookies: true,
+      headers: {
+        "Authorization": "Basic " + Buffer.from(accountData.username + ":" + accountData.app_password).toString("base64")
+      }
+    });
+
+    if (res.error || res.response.statusCode >= 300) {
+      return Promise.reject(
+        this.createPostResponse({
+          error: res.error,
+          message: 'WordPress post failure',
+          additionalInfo: res.body,
+        }),
+      );
+    }
+
+    return this.createPostResponse({ source: JSON.parse(res.body).guid.rendered, additionalInfo: res.body });
+  }
+
+  async postFileSubmission(
+    cancellationToken: CancellationToken,
+    data: FilePostData<WordPressFileOptions>,
+    accountData: WordPressAccountData,
+  ): Promise<PostResponse> {
+    this.checkCancelled(cancellationToken);
+
+    // media upload
+    const fileRes = await Http.post<any>(accountData.instance + '/wp-json/wp/v2/media', '', {
+      data: data.primary,
+      type: 'multipart',
+      skipCookies: true,
+      headers: {
+        "Authorization": "Basic " + Buffer.from(accountData.username + ":" + accountData.app_password).toString("base64")
+      }
+    });
+
+    if (fileRes.error || fileRes.response.statusCode >= 300) {
+      return Promise.reject(
+        this.createPostResponse({
+          error: fileRes.error,
+          message: 'WordPress file upload failure',
+          additionalInfo: fileRes.body,
+        }),
+      );
+    }
+
+    const postres = await Http.post<any>(accountData.instance + '/wp-json/wp/v2/posts', '', {
+      data: this.buildDescriptionPayload(data, accountData, JSON.parse(fileRes.body).id),
+      type: 'multipart',
+      skipCookies: true,
+      headers: {
+        "Authorization": "Basic " + Buffer.from(accountData.username + ":" + accountData.app_password).toString("base64")
+      }
+    });
+
+    if (postres.error || postres.response.statusCode >= 300) {
+      return Promise.reject(
+        this.createPostResponse({
+          error: postres.error,
+          message: 'WordPress posting failure',
+          additionalInfo: postres.body,
+        }),
+      );
+    }
+
+    return this.createPostResponse({
+      source: JSON.parse(postres.body).guid.rendered,
+      additionalInfo:
+        '--- FILE UPLOAD REQUEST ---\n' +
```

Comment:
nit: preferable to store as object { fileRes: fileRes.body, postRes: postRes.body }